plugins/CoreUpdater/Updater.php

```diff
@@ -112,19 +117,58 @@ public function updatePiwik($https = true)
             $this->verifyDecompressedArchive($extractedArchiveDirectory);
             $messages[] = $this->translator->translate('CoreUpdater_VerifyingUnpackedFiles');
 
-            $disabledPluginNames = $this->disableIncompatiblePlugins($newVersion);
-            if (!empty($disabledPluginNames)) {
-                $messages[] = $this->translator->translate('CoreUpdater_DisablingIncompatiblePlugins', implode(', ', $disabledPluginNames));
-            }
+            // we need to load the marketplace already here, otherwise it will use the new, updated file in Piwik 3
+            $marketplace = new MarketplaceApiClient();
+            require_once PIWIK_DOCUMENT_ROOT . '/plugins/CorePluginsAdmin/PluginInstaller.php';
+            require_once PIWIK_DOCUMENT_ROOT . '/plugins/CorePluginsAdmin/MarketplaceApiException.php';
```

Comment:
Because Piwik 3 does not include those files yet, or may not even have those files later, we need to embed the files before the new files are copied. After updating to Piwik 3 those files might otherwise not exist